Traceback (most recent call last):
  File "/usr/local/Caskroom/mambaforge/base/lib/python3.10/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/usr/local/Caskroom/mambaforge/base/lib/python3.10/site-packages/nbclient/client.py", line 1319, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/usr/local/Caskroom/mambaforge/base/lib/python3.10/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
  File "/usr/local/Caskroom/mambaforge/base/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/usr/local/Caskroom/mambaforge/base/lib/python3.10/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/usr/local/Caskroom/mambaforge/base/lib/python3.10/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/usr/local/Caskroom/mambaforge/base/lib/python3.10/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
s2 = df.copy()

# Order x by total Percent (descending)
order = (
    s2.groupby("QID20", as_index=False)["Percent"].sum()
      .sort_values("Percent", ascending=False)["QID20"]
)
s2["QID20"] = pd.Categorical(s2["QID20"], categories=order, ordered=True)


fig = px.bar(
    s2.sort_values("QID20"),
    x="QID20",
    y="Percent",
    color="Respondent Type",
)

fig.update_layout(
    barmode="stack",
    xaxis_title=""
)
fig.write_html('_static/os_valuable_ro.html', full_html=False, include_plotlyjs='cdn')
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mKeyError[0m                                  Traceback (most recent call last)
Cell [0;32mIn[3], line 5[0m
[1;32m      1[0m s2 [38;5;241m=[39m df[38;5;241m.[39mcopy()
[1;32m      3[0m [38;5;66;03m# Order x by total Percent (descending)[39;00m
[1;32m      4[0m order [38;5;241m=[39m (
[0;32m----> 5[0m     [43ms2[49m[38;5;241;43m.[39;49m[43mgroupby[49m[43m([49m[38;5;124;43m"[39;49m[38;5;124;43mQID20[39;49m[38;5;124;43m"[39;49m[43m,[49m[43m [49m[43mas_index[49m[38;5;241;43m=[39;49m[38;5;28;43;01mFalse[39;49;00m[43m)[49m[43m[[49m[38;5;124;43m"[39;49m[38;5;124;43mPercent[39;49m[38;5;124;43m"[39;49m[43m][49m[38;5;241m.[39msum()
[1;32m      6[0m       [38;5;241m.[39msort_values([38;5;124m"[39m[38;5;124mPercent[39m[38;5;124m"[39m, ascending[38;5;241m=[39m[38;5;28;01mFalse[39;00m)[[38;5;124m"[39m[38;5;124mQID20[39m[38;5;124m"[39m]
[1;32m      7[0m )
[1;32m      8[0m s2[[38;5;124m"[39m[38;5;124mQID20[39m[38;5;124m"[39m] [38;5;241m=[39m pd[38;5;241m.[39mCategorical(s2[[38;5;124m"[39m[38;5;124mQID20[39m[38;5;124m"[39m], categories[38;5;241m=[39morder, ordered[38;5;241m=[39m[38;5;28;01mTrue[39;00m)
[1;32m     11[0m fig [38;5;241m=[39m px[38;5;241m.[39mbar(
[1;32m     12[0m     s2[38;5;241m.[39msort_values([38;5;124m"[39m[38;5;124mQID20[39m[38;5;124m"[39m),
[1;32m     13[0m     x[38;5;241m=[39m[38;5;124m"[39m[38;5;124mQID20[39m[38;5;124m"[39m,
[1;32m     14[0m     y[38;5;241m=[39m[38;5;124m"[39m[38;5;124mPercent[39m[38;5;124m"[39m,
[1;32m     15[0m     color[38;5;241m=[39m[38;5;124m"[39m[38;5;124mRespondent Type[39m[38;5;124m"[39m,
[1;32m     16[0m )

File [0;32m/usr/local/Caskroom/mambaforge/base/lib/python3.10/site-packages/pandas/core/groupby/generic.py:1951[0m, in [0;36mDataFrameGroupBy.__getitem__[0;34m(self, key)[0m
[1;32m   1944[0m [38;5;28;01mif[39;00m [38;5;28misinstance[39m(key, [38;5;28mtuple[39m) [38;5;129;01mand[39;00m [38;5;28mlen[39m(key) [38;5;241m>[39m [38;5;241m1[39m:
[1;32m   1945[0m     [38;5;66;03m# if len == 1, then it becomes a SeriesGroupBy and this is actually[39;00m
[1;32m   1946[0m     [38;5;66;03m# valid syntax, so don't raise[39;00m
[1;32m   1947[0m     [38;5;28;01mraise[39;00m [38;5;167;01mValueError[39;00m(
[1;32m   1948[0m         [38;5;124m"[39m[38;5;124mCannot subset columns with a tuple with more than one element. [39m[38;5;124m"[39m
[1;32m   1949[0m         [38;5;124m"[39m[38;5;124mUse a list instead.[39m[38;5;124m"[39m
[1;32m   1950[0m     )
[0;32m-> 1951[0m [38;5;28;01mreturn[39;00m [38;5;28;43msuper[39;49m[43m([49m[43m)[49m[38;5;241;43m.[39;49m[38;5;21;43m__getitem__[39;49m[43m([49m[43mkey[49m[43m)[49m

File [0;32m/usr/local/Caskroom/mambaforge/base/lib/python3.10/site-packages/pandas/core/base.py:244[0m, in [0;36mSelectionMixin.__getitem__[0;34m(self, key)[0m
[1;32m    242[0m [38;5;28;01melse[39;00m:
[1;32m    243[0m     [38;5;28;01mif[39;00m key [38;5;129;01mnot[39;00m [38;5;129;01min[39;00m [38;5;28mself[39m[38;5;241m.[39mobj:
[0;32m--> 244[0m         [38;5;28;01mraise[39;00m [38;5;167;01mKeyError[39;00m([38;5;124mf[39m[38;5;124m"[39m[38;5;124mColumn not found: [39m[38;5;132;01m{[39;00mkey[38;5;132;01m}[39;00m[38;5;124m"[39m)
[1;32m    245[0m     ndim [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39mobj[key][38;5;241m.[39mndim
[1;32m    246[0m     [38;5;28;01mreturn[39;00m [38;5;28mself[39m[38;5;241m.[39m_gotitem(key, ndim[38;5;241m=[39mndim)

[0;31mKeyError[0m: 'Column not found: Percent'

